packages/next/src/server/dev/on-demand-entry-handler.ts

```diff
@@ -452,10 +452,15 @@ export function onDemandEntryHandler({
     return pagePaths
   }
 
+  const doneBuilding = (stats: webpack.Stats) => {
+    const compilationName = stats.compilation.name as any as CompilerNameValues
+    invalidator.doneBuilding([compilationName])
+  }
+  for (const compiler of multiCompiler.compilers) {
+    compiler.hooks.done.tap('NextJsOnDemandEntries', doneBuilding)
+  }
```

Comment:
The compiler is named the same as the compilation. So something like this should work too:

```suggestion
  for (const compiler of multiCompiler.compilers) {
    compiler.hooks.done.tap('NextJsOnDemandEntries', () => invalidator.doneBuilding([compiler.name]))
  }
```

Reply:
Updated in https://github.com/vercel/next.js/pull/46526/commits/0daf3e7591ef60f1a9675dd04bc8d33af7c5423a